### PR TITLE
Trim `lightning:` from LNInvoice bolt11 response

### PR DESCRIPTION
**Pull request by dangeross**
Trims the "lightning:" from the bolt11 field of LNInvoice if one is passed.

```
sdk> parse lightning:lnbc210n1pj3z6d2pp52wuh79egvd3x67zucx48ktkgfmjxa87x4g34q2t8k08gs3g0kjvqdqqcqzzsxqrrssrzjqvgptfurj3528snx6e3dtwepafxw5fpzdymw9pj20jj09sunnqmwpapyqqqqqqq005qqqqlgqqqqqqgq9qsp5x0jl8kwgshnv4mlmapvnfwzm0djs26n8fz0mpn57z0vh9swqqd2s9qyyssqzcxszyy6p6vyvp763pjg3tpl4suxalmm45tus07uu2e53vm27k9rgec4rca725rygurg9rexhkke0h62ewtfpmd4aevz89enaxfzr5gq4pwer8
```

Before
```
{
  "Bolt11": {
    "invoice": {
      "bolt11": "lightning:lnbc210n1pj3z6d2pp52wuh79egvd3x67zucx48ktkgfmjxa87x4g34q2t8k08gs3g0kjvqdqqcqzzsxqrrssrzjqvgptfurj3528snx6e3dtwepafxw5fpzdymw9pj20jj09sunnqmwpapyqqqqqqq005qqqqlgqqqqqqgq9qsp5x0jl8kwgshnv4mlmapvnfwzm0djs26n8fz0mpn57z0vh9swqqd2s9qyyssqzcxszyy6p6vyvp763pjg3tpl4suxalmm45tus07uu2e53vm27k9rgec4rca725rygurg9rexhkke0h62ewtfpmd4aevz89enaxfzr5gq4pwer8",
      "payee_pubkey": "0203e1b9715f923b4e30ccca52b8045870f95cc2fcf7f0b0f66d2ecac29f069e8e",
       ...
    }
  }
}
```
After
```
{
  "Bolt11": {
    "invoice": {
      "bolt11": "lnbc210n1pj3z6d2pp52wuh79egvd3x67zucx48ktkgfmjxa87x4g34q2t8k08gs3g0kjvqdqqcqzzsxqrrssrzjqvgptfurj3528snx6e3dtwepafxw5fpzdymw9pj20jj09sunnqmwpapyqqqqqqq005qqqqlgqqqqqqgq9qsp5x0jl8kwgshnv4mlmapvnfwzm0djs26n8fz0mpn57z0vh9swqqd2s9qyyssqzcxszyy6p6vyvp763pjg3tpl4suxalmm45tus07uu2e53vm27k9rgec4rca725rygurg9rexhkke0h62ewtfpmd4aevz89enaxfzr5gq4pwer8",
      "payee_pubkey": "0203e1b9715f923b4e30ccca52b8045870f95cc2fcf7f0b0f66d2ecac29f069e8e",
       ...
    }
  }
}
```

### CODE REVIEW

File: libs/sdk-core/src/invoice.rs [hunk 1]
```diff
@@ -172,7 +172,7 @@ pub fn parse_invoice(bolt11: &str) -> Result<LNInvoice> {
     let converted_hints = invoice_hints.iter().map(RouteHint::from_ldk_hint).collect();
     // return the parsed invoice
     let ln_invoice = LNInvoice {
-        bolt11: bolt11.to_string(),
+        bolt11: invoice.to_string(),
```
Comment (JssDWt) [inline libs/sdk-core/src/invoice.rs:175]: The resulting invoice string can be slightly different from the actual bolt11 invoice string passed as input by parsing and re-encoding.
I think it's best to do something like:

```rust
let bolt11 = bolt11
        .strip_prefix("lightning:")
        .unwrap_or(bolt11);
let signed = bolt11.parse::<SignedRawInvoice>()?;
```

And pass `bolt11` here.

**Review by JssDWt — CHANGES_REQUESTED** [review]

File: libs/sdk-core/src/invoice.rs [hunk 2]
```diff
@@ -147,10 +147,8 @@ pub fn add_lsp_routing_hints(
 
 /// Parse a BOLT11 payment request and return a structure contains the parsed fields.
 pub fn parse_invoice(bolt11: &str) -> Result<LNInvoice> {
-    let signed = bolt11
-        .strip_prefix("lightning:")
-        .unwrap_or(bolt11)
-        .parse::<SignedRawInvoice>()?;
+    let bolt11 = bolt11.strip_prefix("lightning:").unwrap_or(bolt11);
```
Comment (roeierez) [inline libs/sdk-core/src/invoice.rs:150]: Shouldn't we also strip LIGHTNING (ignore case)?

**Review by roeierez — CHANGES_REQUESTED** [review]

File: libs/sdk-core/src/invoice.rs (hunk 2 above)
Comment (dangeross) [reply libs/sdk-core/src/invoice.rs]: Changed to a case insensitive replace

**Review by roeierez — APPROVED** [review]
LGTM

**Review by JssDWt — APPROVED** [review]
LGTM

**Pull request merged**